flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/abilities/sink/BucketingSpec.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.abilities.sink;
+
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonTypeName;
+
+/**
+ * No properties. This only checks whether the interface is implemented again during deserialization
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonTypeName("Bucketing")
```

Comment:
what is this property good for? other spec don't have it

Reply:
Are you referring to `@JsonTypeName("Bucketing")` or something else?

I'll admit that I copied this class directly from FLIP-376.  `PartitioningSpec` does have a `JsonTypeName`.  

(If you are referring to `JsonIgnoreProperties`, that is in plenty of other classes.  I could imagine removing it from this class since we only expect to use this as a marker interface.  Thoughts?)

Reply:
> Are you referring to @JsonTypeName("Bucketing") or something else?

Yes, I also don't recall why I added it. Might be a mistake or feedback that I got during ML discussion. In any case we should only add code if we know what it is doing (and we want it that way).

Reply:
Ah, figured it out... `@JsonTypeName("Bucketing")` overrides using the class name.

We either get 

```
        "type" : "BucketingSpec"
```
or        
```
        "type" : "Bucketing"
```        
In the compiled plan.

I think that the override makes sense.